design-docs/011-ref-and-ancestry-entry-point.md

```diff
@@ -0,0 +1,220 @@
+# Single Entry Point Object for Refs and Ancestry
+
+This designs tries to solve multiple issues at once,
+see [Notes Towards an Icechunk 2.0Single](./010-notes-towards-an-IC-2.md)
+
+## Design
+
+We add a new object at the root of the repo file structure: `$ROOT/repo`.
+The new object is a flatbuffer with the following structure:
+
+```flatbuffer
+table Tag {
+    name: string (required);
+    snapshot: ObjectId12 (required);
+}
+
+table Branch {
+    name: string (required);
+    snapshot: ObjectId12 (required);
+}
+
+table MetadataItem {
+  name: string (required);
+  value: [uint8] (required);
+}
+
+table SnapshotInfo {
+    id: ObjectId12 (required);
+    parent_offset: u32;
+    flushed_at: uint64 (required);
+    message: str (required);
+    metadata: [MetadataItem] (required);
+}
+
+enum RepoAvailability : ubyte { Online = 0, ReadOnly, Offline }
+
+table RepoStatus {
+    availability: RepoAvailability (required);
+    limited_availability_reason: string;
+    set_at: uint64 (required);
```

Comment:
Also `set_by` perhaps?

Reply:
I think that can go in the reason maybe? I'm not sure what OS users would put it `set_by`